client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/InteractionSection/InteractionEventForm.tsx

```diff
@@ -0,0 +1,235 @@
+    idNumber: '',
+    contactType: -1,
+    creationTime: new Date(),
+    involvedContact: null,
+};
+
+const InteractionEventForm: React.FC<InteractionEventFormProps> = (
+    { onPlaceSubTypeChange, isVisible, interactionData, isNewInteraction }: InteractionEventFormProps): JSX.Element => {
+    const {control, watch, clearErrors, setValue, errors, setError} = useFormContext();
+
+    const placeType = watch(InteractionEventDialogFields.PLACE_TYPE);
+    const placeSubType = watch(InteractionEventDialogFields.PLACE_SUB_TYPE);
+    const interactionStartTime = watch(InteractionEventDialogFields.START_TIME);
+    const interactionEndTime = watch(InteractionEventDialogFields.END_TIME);
+    const isUnknownTime = watch(InteractionEventDialogFields.UNKNOWN_TIME);
+    const locationAddress = watch(InteractionEventDialogFields.LOCATION_ADDRESS);
+    const placeName = watch(InteractionEventDialogFields.PLACE_NAME);
+    const placeDescription = watch(InteractionEventDialogFields.PLACE_DESCRIPTION);
+
+    const [startTime, setStartTime] = useState<Date | null>(isNewInteraction ? null : interactionStartTime);
+    const [endTime, setEndTime] = useState<Date | null>(isNewInteraction ? null : interactionEndTime);
+
+    const formConfig = useMemo(() => getOptionsByPlaceAndSubplaceType(placeType, placeSubType), [placeType, placeSubType]);
+
+    const classes = useStyles();
+    const formClasses = useFormStyles();
+
+    const handleTimeChange = (currentTime: Date, interactionDate: Date, fieldName: string) => {
+        if (isValid(currentTime)) {
+            let newDate = new Date(interactionDate.getTime());
+
+            newDate.setHours(currentTime.getHours());
+            newDate.setMinutes(currentTime.getMinutes());
+
+            if (newDate.getTime()) {
+                clearErrors(fieldName);
+                setValue(fieldName, newDate);
+            }
+        } else {
+            setError(fieldName, {type: 'manual', message: 'שעה לא תקינה'});
+        }
+    }
+
+    const externalizationErrorMessage = useMemo<string>(() => {
+        const initialMessage = '*שים לב כי לא ניתן להחצין מקום אם ';
+        const isPrivatePlace = placeType === placeTypesCodesHierarchy.privateHouse.code;
+        const isTransportationPlace = placeType === placeTypesCodesHierarchy.transportation.code;
+        const externalizationErrors: string[] = [];
+
+        if (isPrivatePlace) {
+            externalizationErrors.push('מדובר בבית פרטי')
+        } else {
+            if (isUnknownTime) {
+                externalizationErrors.push('הזמן אינו ידוע');
+            }
+            if (!isTransportationPlace && !(locationAddress && (placeName || placeDescription))) {
+                externalizationErrors.push('חסרה כתובת ובנוסף חסר שם המוסד או פירוט');
+            }
+        }
+        if (externalizationErrors.length === 0) {
+            return '';
+        } else {
+            setValue(InteractionEventDialogFields.EXTERNALIZATION_APPROVAL, null);
```

Comment:
shouldnt you export this to useEffect triggered by externalizationErrorMessage ?

Reply:
same